main.go

```diff
@@ -1,44 +1,76 @@
 package main
 
 import (
+	"flag"
 	"time"
-
+	"gossip-demo/demo"
 	"github.com/stefankopieczek/gossip/log"
 )
 
 var (
 	// Caller parameters
-	caller = &endpoint{
-		displayName: "Ryan",
-		username:    "ryan",
-		host:        "192.168.0.2",
-		port:        5060,
-		transport:   "UDP",
+	caller = &demo.EndPoint{
+		DisplayName: "Test caller",
+		UserName:    "ted",
+		Host:        "127.0.0.1",
+		Port:        5061,
+		Transport:   "UDP",
 	}
 
 	// Callee parameters
-	callee = &endpoint{
-		displayName: "Ryan's PC",
-		username:    "stefan",
-		host:        "192.168.0.9",
-		port:        5060,
-		transport:   "UDP",
+	callee = &demo.EndPoint{
+		DisplayName: "Test Service",
+		UserName:    "service",
+		Host:        "127.0.0.1",
+		Port:        5060,
+		Transport:   "UDP",
 	}
 )
 
-func main() {
+
+
+func TestCall() {
 	log.SetDefaultLogLevel(log.DEBUG)
 	err := caller.Start()
 	if err != nil {
 		log.Warn("Failed to start caller: %v", err)
 		return
 	}
+	caller.Invite(callee)
+
+}
+
+
+func TestServer() {
+	log.SetDefaultLogLevel(log.DEBUG)
+	err := callee.Start()
+	if err != nil {
+		log.Warn("Failed to start caller: %v", err)
+		return
+	}
 
 	// Receive an incoming call.
-	caller.ServeInvite()
+	callee.ServeInvite()
 
 	<-time.After(2 * time.Second)
 
 	// Send the BYE
-	caller.Bye(callee)
+	callee.Bye(caller)
+}
+
+/*Usage: In one shell window run "go run main.go" without arguments
+  to start server.
+
+  In another shell window, run "go run main.go -call 1" to do a test call
+  */
+
+func main() {
+	var doCall = 0
+	flag.IntVar(&doCall, "call",0, "do a test call")
```

Comment:
Can we change this to be two `flag.Bool`s called "-server" and "-client",  or "-caller" and "-callee"?  I think that would be a bit clearer.